core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandler.java

```diff
@@ -188,6 +190,7 @@ protected CqlRequestHandler(
 
     this.throttler = context.getRequestThrottler();
     this.throttler.register(this);
+    this.requestLogFormatter = new RequestLogFormatter(context);
```

Comment:
@adutra You commented in #1178 that the QueryLogger should be made static to avoid object instantiation on the hot path. I believe this RequestLogFormatter is on a similar hot path here. However, I'm not sure if there is an easy way to avoid it as the RequestLogFormatter needs an instance of DriverContext, which is only set when this request handler is instantiated. Ideas?

Reply:
You are absolutely right. I see that `RequestLogFormatter` only needs the context to access the codec registry and the protocol version. So I see two options: 

1. Do nothing, maybe it's not that bad to instantiate that component at every request; or
2. Create another constructor in `RequestLogFormatter` that takes a codec registry and a protocol version, and use default values for both (this could break if the user is using custom types).

In the end, I think I'd go with option 1.